### PR TITLE
feat(components): [select]add collapse-tags-tooltip

### LINKED ISSUE #6024: [Feature Request] select, Cascader, 希望可以提供个能力, 将选中项置顶, 按选中先后排序
### Existing Component
Yes

### Component Name
select, Cascader

### Description
大数据场景, 客户经常不知道自己选择了什么, 所以希望可以有个置顶的能力, 麻烦了, 非常感谢你们~

### PR DESCRIPTION
**Pull request by Alanscut**
fix: #6024 

display all tags when mouse hover on the text of collapse-tags by use `collapse-tags-tooltip`.

https://user-images.githubusercontent.com/21104054/155279226-9abb54ee-55df-4ee0-8aa9-345e7db669d5.mp4

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by btea** [comment]
image

Maybe the tooltip position should be updated after closing the tag, or like `Naive UI`, fix the tooltip width the same as the input?

**Comment by JeremyWuuuuu** [comment]
@Alanscut @btea Don't you guys think that this is kind of redundant action? Do you have any live example how the other component library did on this case?

**Comment by btea** [comment]
@JeremyWuuuuu `Naive UI` has this case, this is a [link](https://codesandbox.io/s/yg8oy0).

**Comment by Alanscut** [comment]
> @Alanscut @btea Don't you guys think that this is kind of redundant action? Do you have any live example how the other component library did on this case?

I understand that the requirement given in the issue is to use collapse-tags when there are a large number of options and when you want the select box to maintain a certain size. But in this case it is very difficult to find and cancel a selected option in the drop-down menu.

**Comment by Alanscut** [comment]
@btea 
The width has been limited

File: packages/components/select/__tests__/select.spec.ts [hunk 1]
```diff
@@ -882,6 +883,56 @@ describe('Select', () => {
...
+    )
+    await wrapper.find('.select-trigger').trigger('click')
+    const options = getOptions()
+
+    options[0].click()
+    await nextTick()
+    options[1].click()
+    await nextTick()
+    options[2].click()
+    await nextTick()
```
Comment (btea) [inline packages/components/select/__tests__/select.spec.ts:928]: There should only be one `await nextTick()`

Comment (Alanscut) [reply packages/components/select/__tests__/select.spec.ts:928]: ok

Comment (btea) [reply packages/components/select/__tests__/select.spec.ts:926]: There seems to be a problem with this modification, the test fails...

Comment (Alanscut) [reply packages/components/select/__tests__/select.spec.ts:928]: ，，，it seems  `await nextTick()`  is required

**Comment by btea** [comment]
@Alanscut 

https://user-images.githubusercontent.com/24516654/156690389-2e0f866c-ee0a-4a23-822a-efa966333324.mp4

**Comment by Alanscut** [comment]
@btea 
that problem has been solved, 
please check it again, thank you

**Review by jw-foss — APPROVED** [review]

**Pull request merged**